### ISSUE TITLE
Encrypt file using sops using a key store in Azure Vault

### ISSUE THREAD

**Issue by Shaun-Harrison** [mozilla/sops#809]
**Errors:**
```
Get Token request returned http error: 400 and server response: {"error":"invalid_resource","error_description":"AADSTS500011: The resource principal named https://vault.azure.net was not found in the tenant named [TENTANT_ID]. This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You might have sent your authentication request to the wrong tenant.
Traceback (most recent call last):
...
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `az cloud set -n AzureChinaCloud` - this doesn't seem to make a difference but is suppose to change the CLI to know its using China Cloud
- Have a key vault created and a key inside with encrypt and decrypt permission 
- `sops --encrypt --azure-kv https://[VAULT_NAME].vault.azure.cn/keys/[KEY_NAME] secrets.yaml`

## Expected Behavior

Instead of authorization using `https://vault.azure.net` it uses `*.azure.cn` so that it can find the correct tenant
## Environment Summary
```
Linux-3.10.0-1062.4.3.el7.x86_64-x86_64-with-centos-7.9.2009-Core
Python 3.6.8
Installer: RPM

azure-cli 2.18.0

```

**Comment by romeomorcia-wipro** [comment]
@Shaun-Harrison have you found any workaround for this?

**Comment by Shaun-Harrison** [comment]
@romeomorcia-wipro I see you've found the other issue I opened, all updates on this are there

**Comment by romeomorcia-wipro** [comment]
@Shaun-Harrison wrote you the workaround that worked for me too!

I think this is something that should be solved by SOPS team.

I'll add the comment also here:

https://github.com/mozilla/sops/blob/38b25bd449619e1d6da20e637702f7c73203aa44/azkv/keysource.go#L54-L81

As you might see here, sops is taking configs from environment variables, this means that by setting the environment variable to the proper value it will make everything works just fine.

Obviously, it would be nice to have this information in the error message!

before running the sops command you must run:
export AZURE_ENVIRONMENT="azurechinacloud"

**Comment by sdhibit** [comment]
I agree this should be solved within sops by inferring the cloud from the az cli context. I work in an org that constantly switches between clouds, and while setting the env variable does solve the problem, it's not a great experience to remember to change this variable _and_ in addition to changing the az cloud.
